transports/quic/src/transport.rs

```diff
@@ -172,6 +173,19 @@ impl<P: Provider> GenTransport<P> {
             }
         }
     }
+
+    fn create_socket(&self, socket_addr: &SocketAddr) -> io::Result<Socket> {
+        let domain = if socket_addr.is_ipv4() {
+            Domain::IPV4
+        } else {
+            Domain::IPV6
+        };
+        let socket = Socket::new(domain, Type::DGRAM, Some(socket2::Protocol::UDP))?;
```

Comment:
Don't we have to set it to non-blocking as well?

Reply:
also a good question. We don't do it with TCP, `quinn` forwards the `std::UdpSocket` [to the executors](https://github.com/quinn-rs/quinn/blob/70e14b5c26b45ee1e3d5dd64b2a184e2d6376880/quinn/src/runtime.rs#L20-L21), which in case of  `tokio` uses `mio` which has [this comment](https://github.com/tokio-rs/mio/blob/236fc31abf58bfcb8ed66b9ae2ce570acb911e77/src/net/udp.rs#L129-L130), so it seems optional
[According to the documentation](https://www.gnu.org/software/libc/manual/html_node/Open_002dtime-Flags.html):
> Macro: int O_NONBLOCK
This prevents open from blocking for a “long time” to open the file. This is only meaningful for some kinds of files, usually devices such as serial ports; when it is not meaningful, it is harmless and ignored. Often, opening a port to a modem blocks until the modem reports carrier detection; if O_NONBLOCK is specified, open will return immediately without a carrier.

so i'd say it's redundant, wdyt?